### PR TITLE
Cross-check status of elements retrieved from workRestrictions list function

### LINKED ISSUE #12041: WorkQueueManager acquiring the same WQE in multiple agents
**Impact of the bug**
WMAgents

**Describe the bug**
As reported in this comment: https://github.com/dmwm/WMCore/issues/11956#issuecomment-2224384239

we found out that some workflows are having duplicate run/lumis because the very same global workqueue element is actually acquired by multiple agents.

There are multiple workflows having this issue, but the one investigated in the issue 11956 is:
`cmsunified_task_EXO-RunIISummer20UL18wmLHEGEN-05236__v1_T_240606_163348_8728`

which had a single WQE, being acquired both by submit7 and vocms0252.
 
**How to reproduce it**
One could try running a workflow in a team name with multiple agents connected to, but I don't think it is easy to reproduce this problem.

**Expected behavior**
A global workqueue element must be pulled down by a single agent. There is a "Negotiating" protocol/status, which is supposed to ensure that each GQE is only acquired by a single agent.
We need to identify the actual flaw and make sure it NEVER happens again.

**Additional context and error message**
Further details in this comment (but not that relevant): https://github.com/dmwm/WMCore/issues/11956#issuecomment-2224367587

and this as well: https://github.com/dmwm/WMCore/issues/11956#issuecomment-2224384239

### PR DESCRIPTION
**Pull request by amaltaro**
Fixes #12041 

#### Status
ready

#### Description
Summary of changes are:
* provide a more clear log for the "saveElements" action, printing what was actually successfully saved;
* when evaluating available work pulled from workqueue `workRestriction` list function, make sure the actual element document status is `Available`, if not, skip it from work acquisition

#### Is it backward compatible (if not, which system it affects?)
YES

#### Related PRs
None

#### External dependencies / deployment changes
None

### CODE REVIEW

**Comment by amaltaro** [comment]
For my own record, this is being tested in vocms0261 and vocms265, using the team name `testbed-vocms0261`

**Comment by amaltaro** [comment]
I am not sure what happened to my test workflow, but here we go again:
amaltaro_ReReco_Run2022C_LumiMask_Aug2024_Val_240730_195648_3218

**Comment by amaltaro** [comment]
I was not able to reproduce this error with my test workflow, even though I managed to get to the exact scenario that I wanted to test, which was:
1) have 2 agents fetching the Available WQE concurrently
2) and getting them to concurrently (at some degree) update the WQE

As expected, the agent that got late to this update hit a CouchDB error:
```
2024-07-30 21:13:04,238:139833399793408:ERROR:WorkQueueBackend:Couch error saving element: "e9033223936c3762115e6f4daca9a4b4", error "conflict", reason "Document update conflict.
"
```

and properly skipped that workflow/WQE.


This made me analyze once again the logs provided in the GH issue, and my next suspicious is that the WQE was updated by vocms0252 **while** submit7 was slicing through the workqueue list function `workRestrictions`. 
In other words, `workRestrictions` list function yields elements in Available, but when it delivered the full document (include_docs=true) it actually provided the agent with an element that was already in the `Negotiating` status, hence not having any document update conflict.

Reproducing it will be hard, as we would likely have to have thousands of elements in the queue. But I think we can add one extra protection to avoid such problems, and hope that this was the actual issue.

**Comment by amaltaro** [comment]
test this please

**Review by anpicci — APPROVED** [review]
Sounds reasonable to me, thank you @amaltaro

**Comment by amaltaro** [comment]
Thank you, Andrea! I am going to backport this fix and the verbose log one and push a patch release in before the weekend.
